ui/src/cloud/actions/demodata.ts

```diff
@@ -46,26 +65,63 @@ export const getDemoDataBuckets = () => async (
   }
 }
 
-export const getDemoDataBucketMembership = (bucketID: string) => async (
+export const getDemoDataBucketMembership = (bucket: DemoBucket) => async (
   dispatch,
   getState: GetState
 ) => {
+  const state = getState()
   const {
     me: {id: userID},
-  } = getState()
+  } = state
+  const {id: orgID} = getOrg(state)
 
   try {
-    await getDemoDataBucketMembershipAJAX(bucketID, userID)
-
+    await getDemoDataBucketMembershipAJAX(bucket.id, userID)
     dispatch(getBuckets())
```

Comment:
why is it required to refresh all buckets here instead of maintaining the state changes of a single bucket? shouldn't this be an await also?

Reply:
added await. made a request for a single bucket